### DISCUSSION TITLE
Prisma and Kubernetes

### DISCUSSION THREAD

**Discussion by IhsenBouallegue** [prisma/prisma#16017] [Q&A] [answered]
Hello 😄,

I have a Nextjs app that uses a PostgreSQL with Prisma. I want to deploy it to a cluster with Kubernetes. However, it doesn't like the fact that the database is empty (no tables). For development, I am using a seeding script, so no problems there.
Only thing I found is that the seeding/migration step can be added to a CI/CD pipeline. But I don't have that. I simply want my database to at least have the tables so that the app can function correctly.

Any ideas how that can be achieved?

**Comment by nurul3101** [comment, marked as answer]
Hey @IhsenBouallegue 👋 
Do you get any specific error message when deploying to Kubernetes?

**Reply by IhsenBouallegue** [reply]
Hey @nurul3101 for that I need to access the container. I couldn't get around to doing that. But my problem is seeding the database upon creation. How would one do that with prisma inside a kubernetes cluster?

**Reply by nurul3101** [reply]
I haven't tried to deploy a Next app on Kubernetes but I was able to find a [guide](https://willschenk.com/articles/2021/next_js_with_prisma/) which runs you through the process, could you have a look and see if it solves the issue for you?

**Reply by IhsenBouallegue** [reply]
Thank you for the reply. I followed that guide before posting. However, what it does is expose the database to the local machine, so that you can connect to it and run ```npx prisma migrate``` manually. I can't do that every time for every database deployment.

Here is what I want to achieve. When the DB instance is being created it should also be migrated (tables created and seeded) all without my intervention. I think this was possible with Prisma V1 (there is even a guide on that in the old docs). But that's a whole different architecture.

**Question:** can Prisma handle this, or should I create some kind of SQL script to do that? (which is really not optimal, because I have to maintain the schemas in two different places)

**Side note:** the nextjs up is working normally as any web app, it just expects the database to have the tables, so it can fetch them.

**Reply by janpio** [reply]
Your challenge is for your deployment to trigger a database migration via `migrate deploy`. Prisma itself can not be aware of your deployments, so you need to take care of that yourself and then tell Prisma what to do in those cases, e.g deploying the migrations and seeding some data.

**Reply by IhsenBouallegue** [reply]
@janpio exactly. But as there was a guide for Prisma V1 on how to do this,I thought maybe someone did something similar and I don't have to reinvent the wheel.
I am looking right now into automatically triggering a script that does all of that. Thanks for the input!

**Reply by janpio** [reply]
Ah, I was not aware of the Prisma 1 context. As that was an actual infrastructure component that you had to deploy next to your own app, that was much more important to get right in the specific way it was needed - for Prisma 2+ you can do it as you would with any other script.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
